File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for(int i = 0; i < n; i++) {
        	String empty1 = changeBinary(Integer.toBinaryString(arr1[i]), n);
        	String empty2 = changeBinary(Integer.toBinaryString(arr2[i]), n);
        	String ans = "";
        	for(int j = 0; j < n; j++) {	
        		if(empty1.charAt(j) == '1' || empty2.charAt(j) == '1') {
        			ans += "#";
        		}
        		else {
        			ans += " ";
        		}
        	}
        	answer[i] = ans;
        }
        
        return answer;
    }
    
    static String changeBinary(String str, int n) {
		
		String pulsZero = "";
		
		if(str.length() == n) {
			return str;
		}
		else {
			for(int i = 0; i<n-str.length(); i++) {
				pulsZero += "0";
			}
			return pulsZero+str;
		}
	}
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    def todo(x,n):
        ist = []
        while x > 1:
            a, b = divmod(x,2)
            ist.append(b)
            x = a
        ist.append(x)
        for i in range(len(ist),n):
            ist.append(0)
        ist.reverse()
        return ist

    for i in range(0, n):
        st = ''
        arr1result = todo(arr1[i],n)
        arr2result = todo(arr2[i],n)

        for i in range(0, n):
            if arr1result[i] == 1 or arr2result[i] == 1:
                st += '#'
            else:
                st += ' '
        answer.append(st)


    return answer